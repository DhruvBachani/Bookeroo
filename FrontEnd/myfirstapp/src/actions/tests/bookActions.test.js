import React from 'react';
import reducer from '../../../src/reducers'
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore([]);

const createBook = () => ({
    type: 'GET_ERRORS'
})

const postAd = () => ({
    type: 'GET_ERRORS'
})

const getAllBooks = () => ({
    type: 'GET_BOOKS'
});

const getAllCategories = () => ({
    type: 'GET_CATEGORIES'
});

const getBook = () => ({
    type: 'GET_BOOK'
});

const search = () => ({
    type: 'GET_BOOKS'
});

const getBookReviews = () => ({
    type: 'GET_REVIEWS'
});

describe('bookActions', () => {
    it('creates GET_ERRORS when a book cannot be created', () => {
        store.dispatch(createBook());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_ERRORS when an ad cannot be created', () => {
        store.dispatch(postAd());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_BOOKS when it is called successfully', () => {
        store.dispatch(getAllBooks());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_CATEGORIES when it is called successfully', () => {
        store.dispatch(getAllCategories());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_BOOK when it is called successfully', () => {
        store.dispatch(getBook());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_BOOKS when a search query is made', () => {
        store.dispatch(search());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_REVIEWS when it is called successfully', () => {
        store.dispatch(getBookReviews());
        expect(store.getActions()).toMatchSnapshot();
    })
})