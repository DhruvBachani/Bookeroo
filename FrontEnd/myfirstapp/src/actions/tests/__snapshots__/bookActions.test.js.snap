// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookActions creates GET_BOOK when it is called successfully 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_BOOKS",
  },
  Object {
    "type": "GET_CATEGORIES",
  },
  Object {
    "type": "GET_BOOK",
  },
]
`;

exports[`bookActions creates GET_BOOKS when a search query is made 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_BOOKS",
  },
  Object {
    "type": "GET_CATEGORIES",
  },
  Object {
    "type": "GET_BOOK",
  },
  Object {
    "type": "GET_BOOKS",
  },
]
`;

exports[`bookActions creates GET_BOOKS when it is called successfully 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_BOOKS",
  },
]
`;

exports[`bookActions creates GET_CATEGORIES when it is called successfully 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_BOOKS",
  },
  Object {
    "type": "GET_CATEGORIES",
  },
]
`;

exports[`bookActions creates GET_ERRORS when a book cannot be created 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
]
`;

exports[`bookActions creates GET_ERRORS when an ad cannot be created 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
]
`;

exports[`bookActions creates GET_REVIEWS when it is called successfully 1`] = `
Array [
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_ERRORS",
  },
  Object {
    "type": "GET_BOOKS",
  },
  Object {
    "type": "GET_CATEGORIES",
  },
  Object {
    "type": "GET_BOOK",
  },
  Object {
    "type": "GET_BOOKS",
  },
  Object {
    "type": "GET_REVIEWS",
  },
]
`;
